playground/src/pages/CodingEnvironment.tsx

```diff
@@ -122,16 +122,15 @@ end`;
    * This sets the output and number of outputs to the default values.
    * Set to 16 now, that should change dynamically.
    */
-
   const emptyOutput = "\n \n \n \n";
   const [output, setOutput] = React.useState(emptyOutput);
 
   /**
    * This handles the number of outputs that are returned to the user.
-   * It is not optimal to fix that to a certain number, but for
-   * the sake of simplicity we do it for now.
+   * At the moment we return the whole stack always. This is not good UX.
+   * But it is needed to use the Output again as Input for the verification step.
```

Comment:
I'm not sure that there's a problem with returning the entire stack. As you said, we need it because the stack output and the overflow addresses are inputs for verification, and we want people to be able to understand how this is working. Maybe the UX improvement would be to show a section for "verifier inputs", which would contain the full stack + overflow_addrs, etc. Then we could still show a result as just the top # of stack outputs. I think we'd also want to add some hover info messages to explain everything for clarity.

I would probably remove this comment but make an issue for thinking about this

Reply:
Indeed, let's get back to it when we talk in-depth about UX